contribs/gmf/src/d3-lidar-profile/lidarprofilePlot.js

```diff
@@ -0,0 +1,320 @@
+    ctx.clearRect(0, 0, width, height);
+
+    svg.select('.y.axis').selectAll('g.tick line')
+      .style('opacity', '0.5')
+      .style('stroke', '#b7cff7');
+
+    that.options.profileConfig.scaleX = tr.rescaleX(sx);
+    that.options.profileConfig.scaleY = tr.rescaleY(sy);
+  }
+
+  const zoom = d3.zoom()
+    .scaleExtent([1, 100])
+    .translateExtent([[0, 0], [width, height]])
+    .extent([[0, 0], [width, height]])
+    .on('zoom', zoomed);
+
+  function zoomEnd() {
+    ctx.clearRect(0, 0, width, height);
+    that.parent_.loader.updateData();
+  }
+
+  zoom.on('end', zoomEnd);
+
+  d3.select('svg#profileSVG')
+    .call(zoom)
+    .on('dblclick.zoom', null);
+
+  d3.select('svg#profileSVG').selectAll('*').remove();
+
+  const svg = d3.select('svg#profileSVG')
+    .attr('width', width + margin.left)
+    .attr('height', height + margin.top + margin.bottom);
+
+  d3.select('svg#profileSVG')
+    .on('mousemove', () => {
+      that.pointHighlight(that);
+    });
+
+
+  const xAxis = d3.axisBottom(sx);
+  const yAxis = d3.axisLeft(sy)
+    .tickSize(-width);
+
+  svg.select('.y.axis').selectAll('g.tick line').style('stroke', '#b7cff7');
+
+  svg.append('g')
+    .attr('class', 'y axis')
+    .call(yAxis);
+
+  svg.append('g')
+    .attr('class', 'x axis')
+    .call(xAxis);
+
+  svg.select('.y.axis').attr('transform', `translate(${margin.left}, ${margin.top})`);
+  svg.select('.x.axis').attr('transform', `translate(${margin.left}, ${height + margin.top})`);
+
+  svg.select('.y.axis').selectAll('g.tick line')
+    .style('opacity', '0.5')
+    .style('stroke', '#b7cff7');
+
+  this.options.profileConfig.previousDomainX = sx.domain();
+  this.options.profileConfig.previousDomainY = sy.domain();
+
+};
+
+/**
+ * Update the Openlayers overlay that displays point position and attributes values
+ * @export
+ * @param {gmf.lidarProfile.plot} that scope of the plot class
+*/
+gmf.lidarProfile.plot.prototype.pointHighlight = function(that) {
+
+  const svg = d3.select('svg#profileSVG');
+  const pointSize = that.options.profileConfig.pointSize;
+  const margin = that.options.profileConfig.margin;
+  const tolerance = that.options.profileConfig.tolerance;
+
+  const canvasCoordinates = d3.mouse(d3.select('#profileCanvas').node());
+  const sx = that.options.profileConfig.scaleX;
+  const sy = that.options.profileConfig.scaleY;
+  let cx, cy;
+  const p = that.utils_.getClosestPoint(that.parent_.loader.profilePoints,
+    canvasCoordinates[0], canvasCoordinates[1], tolerance);
+
+  if (p != undefined) {
+
+    cx = sx(p.distance) + margin.left;
+    cy = sy(p.altitude) + margin.top;
+
+    svg.selectAll('#highlightCircle').remove();
+
+    d3.select('svg#profileSVG').append('circle')
+      .attr('id', 'highlightCircle')
+      .attr('cx', cx)
+      .attr('cy', cy)
+      .attr('r', pointSize + 1)
+      .style('fill', 'orange');
+
+    const html = `Distance: ${Math.round(10 * p.distance) / 10}<br>
```

Comment:
I don't know exactly how to do this now, but can you start to move this into an function (getTooltipHtml or something like that ?). We must be able to translate texts inside.